app/services/jwt_token.rb

```diff
@@ -23,10 +23,12 @@ def iat
 
   def decode
     return nil if token.blank?
-    return nil if secret.blank?
-
-    payload = JWT.decode(token, secret.token)
-    payload.first
+    if secret.blank?
+      payload = JWT.decode(token, nil, false)
```

Comment:
This introduces a huge security gap. With this everybody who can create a JWT can login as every user.

Reply:
I can't stress this enough: This basically disables authentication. Anybody can login as any user.

Reply:
Yes, I had made a note of this the first time when you mentioned it. We can invoke the token introspection endpoint on KeyCloak which will take care of the validation of the token and I will also additionally verify a few claims and check the signature so that we are good with the security concerns.